apps/api-gateway/src/auth/auth.service.ts

```diff
@@ -0,0 +1,124 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { JwtService } from '@nestjs/jwt';
+import { JwtPayload } from 'jsonwebtoken';
+import { firstValueFrom } from 'rxjs';
+import { User } from '../user/entities/user.entity';
+import { UserService } from '../user/user.service';
+import { Payload } from './types/jwt.type';
+import { GithubUser } from './types/user.type';
+
+type LoginReturn = Partial<GithubUser> & { access_token?: string; refresh_token?: string } & {
+  isExist: boolean;
+};
+
+@Injectable()
+export class AuthService {
+  public constructor(
+    private readonly jwtService: JwtService,
+    private readonly configService: ConfigService,
+    private readonly userService: UserService,
+  ) {}
+
+  public async login(user: GithubUser): Promise<LoginReturn> {
+    const { email, githubId, name } = user;
+
+    const existUser = await firstValueFrom(await this.userService.loginOrCreateUser(email, name));
+
+    if (existUser.data.isExist === false) {
+      // * 유저가 존재하지 않을경우 회원 가입 처리를 위해 최소 데이터만 반환
+      return {
+        email: existUser.data.email,
+        githubId,
+        name: existUser.data.nickname,
+        isExist: false,
+      };
+    }
+
+    const paylaod: JwtPayload = { email, sub: githubId };
+
+    return {
+      access_token: this.jwtService.sign(paylaod, this.configService.get('jwt.accessSignOptions')),
+      refresh_token: this.jwtService.sign(paylaod, {
+        ...this.configService.get('jwt.refreshSignOptions'),
+        secret: this.configService.get('jwt.secret'),
+      }),
+      ...existUser.data,
+    };
+  }
+
+  public async refreshAccessToken(refreshToken: string) {
+    try {
+      const payload = this.jwtService.verify(refreshToken, {
+        secret: this.configService.get('jwt.secret'),
+      });
+      const user = await firstValueFrom(await this.validateUser(payload.email));
+
+      if (user.status === 200) {
+        return this.generateAccessToken(payload);
+      }
+    } catch (err) {
+      return null;
```

Comment:
에러 객체를 사용하지 않는다면 작성하지 않는게 좋을 것 같습니다~
```suggestion
    } catch {
      return null;
```

Reply:
[b23380b](https://github.com/GrebbMe/grabbme-be/pull/53/commits/b23380b73c41ba56073479f1018a2672db2d10a8)
위의 커밋에 반영하였습니다!